packages/devtools_app/lib/src/http/_http_date.dart

```diff
@@ -221,11 +221,7 @@ class HttpDate {
 
     int expectNum(String separator) {
       int pos;
-      if (separator.length > 0) {
-        pos = date.indexOf(separator, index);
-      } else {
-        pos = date.length;
-      }
+      pos = separator.length > 0 ? date.indexOf(separator, index) : date.length;
```

Comment:
changes like this seem like an improvement. Fyi @pq have we thought about adding a lint for preferring conditionals?
The one in Dart Code Metrics goes a bit overboard. I'd imagine if we added it to the core linter it would only fire for short expressions like this and not the large ones that code metrics suggests.

Reply:
what about `isNotEmpty` for `seperator`?

Reply:
I'd also be happy with a lint + fix for isEmpty and isNotEmpty.  Seems like a nice simple lint to nudge people to use the Dart core libraries. Looks like our users use `.isNotEmpty` about 10X as often then `.length > 0` so the lint shouldn't be contreversial.

Reply:
We HAVE that lint. Somewhere...

Reply:
https://dart-lang.github.io/linter/lints/prefer_is_not_empty.html – HA! We should expand this to String!

Reply:
https://github.com/dart-lang/linter/issues/3889

Reply:
Related: [`avoid_bool_literals_in_conditional_expressions`](https://dart-lang.github.io/linter/lints/avoid_bool_literals_in_conditional_expressions.html).